src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/MultiPloidyGenotyperCacheUnitTest.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
+
+import htsjdk.samtools.util.OverlapDetector;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.broadinstitute.hellbender.tools.copynumber.formats.records.SimpleCount;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.MinimalGenotypingEngine;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.StandardCallerArgumentCollection;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.genotyper.SampleList;
+import org.jetbrains.annotations.NotNull;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.*;
+
+public class MultiPloidyGenotyperCacheUnitTest extends GATKBaseTest {
+
+
+    @DataProvider
+    public static Object[][] getGenotyperParams() {
+        return new Object[][]{
+                {"1:10-30", 2},
+                {"1:70-85", 3},
+                {"1:80-93", 3},
+              //  {"1:89-92", 2},  TODO is this a bug?
```

Comment:
So I think we have a bug.  Maybe it's not in the original code but I'm thinking it probably is.  If you have a region which contains both default ploidy 2 and a special ploidy 1 region then you end up calling that at ploidy 1 because it's picking the highest of the ALTERNATE ploidies, not the highest of any overlapping ploidy.

Reply:
@jamesemery @rickymagner

Reply:
I think the case Louis was referring to was a hypothetical user-provided bed file that looked like:
```
chr1   1000   1700   1
chr1   1500   2000   2
```
Then maybe the current behavior would be to still use ploidy 1 in the 1500-1700 region even though the "2" overlaps there, because the 2 is the built-in default? Is this correct Louis?

Reply:
```
   private int getPloidyToUseAtThisSite(Locatable region) {
       return ploidyRegions.getOverlaps(region)
               .stream()
               .mapToInt(SimpleCount::getCount)
               .max()
               .orElse(defaultPloidy);
```               
Am I wrong in interpreting that in that case this would fall into the `max()` command? There is a simple-count object for both `chr1   1500   2000   2` and `chr1   1000   1700   1` and it would fall into 2. The only time default ever matters is if there is nothing specified that covers the site?

Reply:
Ricky and I chatted and it seems like it's working as intended.  Used the explicitly specified ploidy if a region overlaps  one even if it includes default region which is higher.